core/src/main/scala/org/apache/spark/rdd/CoalescedRDD.scala

```diff
@@ -192,7 +192,8 @@ private class PartitionCoalescer(maxPartitions: Int, prev: RDD[_], balanceSlack:
     def resetIterator(): Iterator[(String, Partition)] = {
       val iterators = (0 to 2).map( x =>
         prev.partitions.iterator.flatMap(p => {
-          if (currPrefLocs(p).size > x) Some((currPrefLocs(p)(x), p)) else None
+          val locs = currPrefLocs(p)
+          if (locs.size > x) Some((locs(x), p)) else None
```

Comment:
@rxin I don't follow your first sentence? 

Second sentence says use size for a collection and Seq is a collection.